src/client/proxy/http.cc

```diff
@@ -130,11 +133,12 @@ void HTTPProxyHandler::AsyncHandleReadHeaders(
       boost::asio::buffers_begin(bufs) + m_Protocol.m_Buffer.size());
 
   if (!m_Protocol.HandleData(tbuffer)) {
-    LOG(debug) << "AsyncHandleRead: error HandleData() " <<  "check http proxy";
+    LOG(debug) << "HTTPProxy: error HandleData() " <<  "check http proxy";
     HTTPRequestFailed();  // calls Terminate
     return;
   }
-  size_t num_additional_bytes = m_Protocol.m_Buffer.size() - bytes_transfered;
+  std::size_t num_additional_bytes = m_Protocol.m_Buffer.size()
+    - bytes_transfered;
```

Comment:
Still doesn't prevent overflow if bytes transferred is greater than buffer size. Ensure that this will never happen.